### ISSUE TITLE
Division by zero error in dashtrends

### ISSUE THREAD

**Issue by lmeyer1** [PrestaShop/PrestaShop#31008]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

https://github.com/PrestaShop/dashtrends/blob/b1f31432caf32ad8979bd20f04a90a4a9d78b84a/dashtrends.php#L274
gives a division by zero error if the `Compare to` date range in the dashboard has the same date as `from` and `to`.
![2023-01-24_095710](https://user-images.githubusercontent.com/7520695/214249677-349ac281-a794-473b-88bd-e483e28d4d71.png)

In PHP 7.x this is only a warning, in PHP 8.x it is an unhandled exception.

### Expected behavior

A date range with the same date as start and end can be considered as a misconfiguration.
But we need to test for that case before running into a exception.

### Steps to reproduce

On an installation using PHP 8.x
1. Configure the dashboard dates as in the screenshot.
2. Using Google Chrome (for dev tools), hit F12, then navigate to the BO dashboard
3. In dev tools, `Network` tab, notice that the call to the `dashtrends` module failed with an Error 500
![2023-01-24_103228](https://user-images.githubusercontent.com/7520695/214257121-a5cc5bd8-04bb-4b96-9efd-c0d54fb4b83d.png)

### PrestaShop version(s) where the bug happened

1.7.8.7

### PHP version(s) where the bug happened

8.x (in versions before 8.0, only a warning is raised)

### If your bug is related to a module, specify its name and its version

dashtrends dev and 2.1.0

**Comment by sallemiines** [comment]
Hello @lmeyer1 

Thanx for reporting the issue.

Could you please provide us how you configured the dashboard dates with  screenshots or a screenrecord please ! 

Thank you !

**Comment by lmeyer1** [comment]
> Could you please provide us how you configured the dashboard dates with screenshots or a screenrecord please !

@sallemiines 
There is already a screenshot of the configuration above.

**Comment by florine2623** [comment]
Hello @lmeyer1 ,

I can't reproduce your issue with PS 8.0.1 and dashtrends v2.1.0.

https://user-images.githubusercontent.com/16019289/214797308-e96e04c0-0c5d-4f69-a4b4-869a731205d4.mov

In my console I don't have any 500 error. Am I missing something ?
Thanks!

**Comment by lmeyer1** [comment]
@florine2623 
In your video, you `Compare with` from `2023-01-01` to `2023-01-31`.
![2023-01-24_095710](https://user-images.githubusercontent.com/7520695/214249677-349ac281-a794-473b-88bd-e483e28d4d71.png)
Please notice here, that the `Compare with` dates are both `2022-12-31`, they are the same. From `2022-12-31` to `2022-12-31`.
This does not make much sense as a configuration, but as I wrote above:

>A date range with the same date as start and end can be considered as a misconfiguration.
But we need to test for that case before running into a exception.

**Comment by florine2623** [comment]
Oh indeed, sorry.

I added the same date range as you. But I still don't see any error :

https://user-images.githubusercontent.com/16019289/214803435-d6bbb9a9-631b-4f3d-bbe6-aec830cadec3.mov

**Comment by lmeyer1** [comment]
Probably the bug only happens, if there is data to compare, i.e. you have an order on `2022-12-31` that is valid and has an invoice date. Not sure about this though.

**Comment by florine2623** [comment]
@lmeyer1 ,

I tested again, I couldn't find any error 500, only 400, 307 and 302.


https://user-images.githubusercontent.com/16019289/214817490-edaa2093-8a75-4072-a850-e4fac7f78ecf.mov

![image](https://user-images.githubusercontent.com/16019289/214818082-de1b4fb0-cd80-4dc9-8378-ca95faf980d9.png)

**Comment by lmeyer1** [comment]
@florine2623
There are two issues with the screen record:
- You didn't show the dashboard itself. Are you able to see the two lines in the graph if the `compare to` is an interval (i.e. `from != to`) ?
- You don't have orders in January, and you are comparing January to the 28.11.2022. I think you wouldn't get any output anyway. If you have no output, you cannot get any error.

If you still are not able to reproduce, please report this to a dev.

**Comment by HanaRebaiQA** [comment]
Hello @lmeyer1 

As I see, you have mentioned the issue with the PS1787 version and php 8.x. 
Or, PS1787 is not compatible with php8.x.

**Comment by lmeyer1** [comment]
@HanaRebaiQA 
I was not very explicit. I wrote above
> in versions before 8.0, only a warning is raised

Using PS 1.7.8.7 with PHP 7.x, there is a PHP warning division by zero. It only appears in the server logs, depending on error log configuration, but they don't show in the navigator.

**Comment by hibatallahAouadni** [comment]
Hello @lmeyer1 

- With PS1787 + PHP 7.4 -> no issue
- With PS800 + PHP 7.4 -> no issue

Could you please confirm with which version of PS and PHP did you encounter this problem?
Waiting for your feedback.

Thanks!

**Comment by lmeyer1** [comment]
@hibatallahAouadni 
The error shows with PHP 8.x.
With PHP 7.x, it shows only in the error log, if we have enabled logging of warnings. So do not care about PHP 7.x, only about PHP 8.x

**Comment by hibatallahAouadni** [comment]
Hello @lmeyer1 

> The error shows with PHP 8.x.

You mean PrestaShop 1.7.8.7 + PHP 8.x?? if so, that's normal cause PHP 8.x is not compatible with PS1787.
Waiting for your feedback.

Thanks!

**Comment by lmeyer1** [comment]
@hibatallahAouadni 
I mean:
- PS 1.7.8.7 + PHP 7.x => error is in the server logs 
- PS 8.0 + PHP 7.x => error is in the server logs 
- PS 8.0 + PHP 8.x => error is in the server logs and in the navigator

**Comment by hibatallahAouadni** [comment]
Hello @lmeyer1 

Despite our efforts, we were not able to reproduce your issue with the provided information. It seems that your issue is not a PrestaShop core bug but most likely a module problem.

We will close this issue.

Thanks!

**Comment by lmeyer1** [comment]
@hibatallahAouadni 
>It seems that your issue is not a PrestaShop core bug but most likely a module problem.

Yes, I reported from beginning as a module problem with `dashtrends`. Maybe there was a misunderstanding.

For the moment, this is a minor problem for me, but when I have time, I will send you a PR to fix, with a more precise way to reproduce. If this happens (that I have time for a PR), I will ask you to reopen this issue. Thanks.

**Comment by stgarofalo81** [comment]
Hi everyone, I also have this problem.
I am using Prestashop 8.2.0 with PHP 8.0.30 MySQL 8.0.30.
I migrated this site from prestashop 1.6.8 to the new Prestashop 8.2.
I used a module to move all the data.
I see the statistics of all the old orders but the new orders that I receive and that are completed are not shown.
What can I do? And the dashtrends module or do I have to work on the DB??

![Image](https://github.com/user-attachments/assets/02e40c64-b989-4f1d-a44a-5120e945f792)
